metadata/layers.go

```diff
@@ -65,6 +66,19 @@ type StackRunImageMetadata struct {
 	Mirrors []string `toml:"mirrors" json:"mirrors,omitempty"`
 }
 
+func (sm *StackMetadata) BestRunImageMirror(registry string) (string, error) {
```

Comment:
Minor: Personally I think it made more sense to be a standalone function instead of it being on the `StackMetadata` but it's only aesthetics.

Reply:
The decision to put it on the StackMetadata was on the back of the idea that it has all the RunImage + Mirrors information so it felt like the right receiver. But as you mentioned, it is an aesthetics thing :)